app/org/sagebionetworks/bridge/dynamodb/DynamoTask.java

```diff
@@ -56,111 +68,149 @@ public TaskStatus getStatus() {
             return TaskStatus.FINISHED;
         } else if (startedOn != null) {
             return TaskStatus.STARTED;
-        } else if (expiresOn != null && DateTime.now().isAfter(expiresOn)) {
-            return TaskStatus.EXPIRED;
-        } else if (scheduledOn != null && DateTime.now().isBefore(scheduledOn)) {
-            return TaskStatus.SCHEDULED;
+        }
+        if (timeZone != null) {
+            DateTime now = DateTime.now(timeZone);
+            if (localExpiresOn != null && now.isAfter(getExpiresOn())) {
+                return TaskStatus.EXPIRED;
+            } else if (localScheduledOn != null && now.isBefore(getScheduledOn())) {
+                return TaskStatus.SCHEDULED;
+            }
```

Comment:
Would it be cleaner to capture getExpiresOn() and getScheduledOn() into local variables, so we can do the null check on the thing we're testing against?

Also, can getExpiresOn() and getScheduledOn() ever not be complete timestamps? DateTime.now(timeZone) is fine, but as long as they're complete timestamps, it shouldn't matter what timezone they're in to compare times.